# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/FetchTypes.ipdlh
# Commit: cf972fe0d961
# Full Hash: cf972fe0d96164788f998f04008930474526e400
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-02-26 21:36:36
# Regressor Bug: 1754365
# File Overlap Count: 1
# Description:
#   Bug 1754365 - Create IPC for transferring NavigationPreload's PerformanceTimingData. r=dom-worker-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138251
# ==============================================================================

diff -r f8afd25bf41e -r cf972fe0d961 dom/fetch/FetchTypes.ipdlh
--- a/dom/fetch/FetchTypes.ipdlh	Sat Feb 26 08:40:25 2022 +0000
+++ b/dom/fetch/FetchTypes.ipdlh	Sat Feb 26 08:40:26 2022 +0000
@@ -5,6 +5,7 @@
 include IPCStream;
 include ChannelInfo;
 include PBackgroundSharedTypes;
+include PerformanceTimingTypes;
 
 include protocol PRemoteLazyInputStream;
 
@@ -100,5 +101,19 @@
   ChildToParentStream? alternativeBody;
 };
 
+struct ParentToParentResponseWithTiming {
+  ParentToParentInternalResponse response;
+  IPCPerformanceTimingData timingData;
+  nsString initiatorType;
+  nsString entryName;
+};
+
+struct ParentToChildResponseWithTiming {
+  ParentToChildInternalResponse response;
+  IPCPerformanceTimingData timingData;
+  nsString initiatorType;
+  nsString entryName;
+};
+
 } // namespace ipc
 } // namespace mozilla